src/lib/ast.rs

```diff
@@ -473,6 +474,46 @@ impl NodeId {
         Ok(())
     }
 
+    /// Attach children to given node
+    pub fn copy_subtree<T: Clone>(&mut self, _to_node: NodeId, position: u16, _arena: &mut Arena<T>) -> ArenaResult {
```

Comment:
Is there a reason why so many argument start with an underscore? Usually we only use the underscore if the variable name is used more than once.